doc/jvm/xgboost_spark_migration.rst

```diff
@@ -0,0 +1,140 @@
+######################################################
+Migration Guide: How to migrate from XGBoost Spark 3.x
+######################################################
+
+XGBoost Spark underwent significant modifications beginning with version 3.0,
+which may cause compatibility issues with existing user code.
+
+This guide will walk you through the process of updating your code to ensure
+it's compatible with XGBoost Spark 3.0 and later versions.
+
+**********************
+XGBoost Spark Packages
+**********************
+
+XGBoost Spark 3.0 introduced a single uber package named xgboost-spark_2.12-3.0.0.jar, which bundles 
+both xgboost4j and xgboost4j-spark. This means you can now simply use `xgboost-spark`` for your application.
+
+.. code-block:: xml
+
+  <dependency>
+      <groupId>ml.dmlc</groupId>
+      <artifactId>xgboost-spark_${scala.binary.version}</artifactId>
+      <version>3.0.0</version>
+  </dependency>
+
+When submitting the XGBoost application to the Spark cluster, you only need to specify the single `xgboost-spark` package.
+
+.. code-block:: bash
+
+  spark-submit \
+    --jars xgboost-spark_2.12-3.0.0.jar \
+    ... \
+
+**************
+XGBoost Ranking
+**************
+
+The ability to handle ranking problems using XGBoostRegressor has been discontinued.
+As an alternative, we have introduced XGBoostRanker, which is specifically designed
+to support ranking algorithms.
+
+.. code-block:: scala
+  
+  // before 3.0
+  val regressor = new XGBoostRegressor().setObjective("rank:ndcg")
+
+  // after 3.0
+  val ranker = new XGBoostRanker()
+
+******************************
+XGBoost Constructor Parameters
+******************************
+
+XGBoost Spark now categorizes parameters into two groups: XGBoost-Spark parameters and XGBoost parameters.
+When constructing an XGBoost estimator, only XGBoost-specific parameters are permitted. XGBoost-Spark specific 
```

Comment:
What's the motivation for this?

Reply:
1. XGBoost-specific parameters can be got from https://xgboost.readthedocs.io/en/stable/parameter.html and we don't have a spark-specific parameters to refer. 
2. (Not a strong reason) The xgboost-specific parameters can be sent to xgboost library without doing filtering, if involving xgboost-spark parameters, we need to add a whiltelist that blocks them to be passed to xgboost library.
3. Spark ML uses setters for all parameters, and don't accept any parameters during constructing.